tests/workfiles2_functions.py

```diff
@@ -0,0 +1,196 @@
+    assert app_dialog.root.tabs["Assets"].exists(), "Assets tab is missing"
+    assert app_dialog.root.tabs["Shots"].exists(), "Shots tab is missing"
+    assert app_dialog.root.tabs["All"].exists(), "All tab is missing"
+    assert app_dialog.root.tabs["Working"].exists(), "Working tab is missing"
+    assert app_dialog.root.tabs["Publishes"].exists(), "Publishes tab is missing"
+
+    # Buttons in both, File Open and File Save dialogs
+    assert app_dialog.root.buttons[
+        "+ New Task"
+    ].exists(), "+ New Task button is missing"
+    assert app_dialog.root.buttons["Cancel"].exists(), "Cancel button is missing"
+
+    # Make sure all text fields are showing up
+    # text fields in both, File Open and File Save dialogs
+    assert app_dialog.root[
+        "Search All Files"
+    ].exists(), "Search All Files text field is missing"
+    assert app_dialog.root[
+        "Search Entity"
+    ].exists(), "Search My Tasks text field is missing"
+
+
+def _test_expected_ui_showing(app_dialog, tab_name):
+    """
+    Basic tabs UI validation to make sure all buttons, tabs and fields are available
+    """
+    # Make sure all expected UI is showing up
+    assert app_dialog.root.captions[
+        "Filter by Pipeline Step"
+    ].exists(), "Pipeline Step filters are missing"
+    assert app_dialog.root.buttons[
+        "Select All"
+    ].exists(), "Pipeline Step filters Select All button is missing"
+    assert app_dialog.root.buttons[
+        "Select None"
+    ].exists(), "Pipeline Step filters Select None button is missing"
+    assert app_dialog.root.checkboxes[
+        "My Tasks Only"
+    ].exists(), "My Tasks Only checkbox is missing"
+    assert app_dialog.root.buttons[
+        "+ New Task"
+    ].exists(), "+ New Task button is missing"
+    assert app_dialog.root[
+        "Search Entity"
+    ].exists(), "Search {0} text field is missing".format(tab_name)
+
+
+def _navigate_and_validate_breadcrumb(app_dialog, selection_hierarchy, entities):
+    """
+    Navigation and breadcrumb validation
+    """
+    # Got to the model task and validate breadcrumb
+
+    for num, item in enumerate(selection_hierarchy):
+
+        app_dialog.root.outlineitems[item].waitExist(timeout=30)
+        if num + 1 != len(selection_hierarchy):
+            # If we are not in the last item in the hierarchy then click the item
+            app_dialog.root.outlineitems[item].mouseDoubleClick()
+
+    # Validate content dialog
+    assert app_dialog.root.cells[
+        selection_hierarchy[1]
+    ].exists(), "{0} is missing in content dialog".format(selection_hierarchy[1])
+    assert app_dialog.root.cells[
+        "{0} - {0}".format(selection_hierarchy[2])
+    ].exists(), "{0} task is missing in content dialog".format(selection_hierarchy[2])
+    assert app_dialog.root.cells[
+        "{0} - {0}".format(entities[3])
+    ].exists(), "{0} task is missing in content dialog".format(entities[3])
```

Comment:
It looks like these checks are not navigating or validating the breadcrumb so perhaps they should be in a different method?